server/src/main/java/org/elasticsearch/common/time/FormatNames.java

```diff
@@ -0,0 +1,124 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.time;
+
+public enum FormatNames {
+    ISO8601("iso8601", "iso8601"),
+    BASICDATE("basicDate", "basic_date"),
+    BASICDATETIME("basicDateTime", "basic_date_time"),
+    BASICDATETIMENOMILLIS("basicDateTimeNoMillis", "basic_date_time_no_millis"),
+    BASICORDINALDATE("basicOrdinalDate", "basic_ordinal_date"),
+    BASICORDINALDATETIME("basicOrdinalDateTime", "basic_ordinal_date_time"),
+    BASICORDINALDATETIMENOMILLIS("basicOrdinalDateTimeNoMillis", "basic_ordinal_date_time_no_millis"),
+    BASICTIME("basicTime", "basic_time"),
+    BASICTIMENOMILLIS("basicTimeNoMillis", "basic_time_no_millis"),
+    BASICTTIME("basicTTime", "basic_t_time"),
+    BASICTTIMENOMILLIS("basicTTimeNoMillis", "basic_t_time_no_millis"),
+    BASICWEEKDATE("basicWeekDate", "basic_week_date"),
+    BASICWEEKDATETIME("basicWeekDateTime", "basic_week_date_time"),
+    BASICWEEKDATETIMENOMILLIS("basicWeekDateTimeNoMillis", "basic_week_date_time_no_millis"),
+    DATE("date", "date"),
+    DATEHOUR("dateHour", "date_hour"),
+    DATEHOURMINUTE("dateHourMinute", "date_hour_minute"),
+    DATEHOURMINUTESECOND("dateHourMinuteSecond", "date_hour_minute_second"),
+    DATEHOURMINUTESECONDFRACTION("dateHourMinuteSecondFraction", "date_hour_minute_second_fraction"),
+    DATEHOURMINUTESECONDMILLIS("dateHourMinuteSecondMillis", "date_hour_minute_second_millis"),
+    DATEOPTIONALTIME("dateOptionalTime", "date_optional_time"),
+    DATETIME("dateTime", "date_time"),
+    DATETIMENOMILLIS("dateTimeNoMillis", "date_time_no_millis"),
+    HOUR("hour", "hour"),
+    HOURMINUTE("hourMinute", "hour_minute"),
+    HOURMINUTESECOND("hourMinuteSecond", "hour_minute_second"),
+    HOURMINUTESECONDFRACTION("hourMinuteSecondFraction", "hour_minute_second_fraction"),
+    HOURMINUTESECONDMILLIS("hourMinuteSecondMillis", "hour_minute_second_millis"),
+    ORDINALDATE("ordinalDate", "ordinal_date"),
+    ORDINALDATETIME("ordinalDateTime", "ordinal_date_time"),
+    ORDINALDATETIMENOMILLIS("ordinalDateTimeNoMillis", "ordinal_date_time_no_millis"),
+    TIME("time", "time"),
+    TIMENOMILLIS("timeNoMillis", "time_no_millis"),
+    TTIME("tTime", "t_time"),
+    TTIMENOMILLIS("tTimeNoMillis", "t_time_no_millis"),
+    WEEKDATE("weekDate", "week_date"),
+    WEEKDATETIME("weekDateTime", "week_date_time"),
+    WEEKDATETIMENOMILLIS("weekDateTimeNoMillis", "week_date_time_no_millis"),
+    WEEKYEAR("weekyear", "week_year"),
+    WEEKYEARWEEK("weekyearWeek", "weekyear_week"),
+    WEEKYEARWEEKDAY("weekyearWeekDay", "weekyear_week_day"),
+    YEAR("year", ""),
```

Comment:
In all the others where camel case and snake case are the same you've duplicated the name, but here there's an empty string.

Reply:
forgot about this. good find!